test/hp/storage/file-test.js

```diff
@@ -0,0 +1,299 @@
+var helpers = require('../../helpers'),
+    async = require('async'),
+    http = require('http'),
+    hock = require('hock'),
+    _ = require('underscore');
+
+var authenticate = function (hockInstance) {
+  hockInstance
+    .post('/v2.0/tokens', {
+      auth: {
+        apiAccessKeyCredentials: {
+          accessKey: 'MOCK-USERNAME',
+          secretKey: 'MOCK-API-KEY'
+        }
+      }
+    })
+    .reply(200, {
+      access: {
+        token: {
+          expires: '2016-12-26T18:25:46Z',
+          id: '4bc7c5dabf3e4a49918683437d386b8a',
+          tenant: {
+            enabled: true,
+            id: '5ACED3DC3AA740ABAA41711243CC6949',
+            name: 'MOCK-USERNAME',
+            description: 'MOCK-USERNAME'
+          }
+        },
+        serviceCatalog: [
+          {
+            endpoints: [
+              {
+                region: 'region-a.geo-1',
+                tenantId: 'HPCloudFS_00aa00aa',
+                publicURL: 'http://localhost:12345/v1/HPCloudFS_00aa00aa',
+                internalURL: 'https://snet-storage101.ord1.clouddrive.com/v1/HPCloudFS_00aa00aa'
+              }
+            ],
+            name: 'swift',
+            type: 'object-store'
+          }
+        ],
+        user: {
+          username: 'MOCK-USERNAME',
+          roles_links: [],
+          id: 'bf3b85477d06430c8044d5b2e5e6dc5f',
+          roles: [],
+          name: 'MOCK-USERNAME'
+        }
+      }
+    });
+};
+
+describe('pkgcloud/openstack/storage/', function () {
+
+  describe('The openstack Storage client', function () {
+
+    describe('copy', function () {
+      var client, hockInstance, authHockInstance, server, authServer;
+
+      beforeEach(function (done) {
+        client = helpers.createClient('hp', 'storage');
```

Comment:
I was focusing on the HP storage API as that is what I am familiar with. I'm not sure if those tests will work with other Openstack clients because they will be expecting different responses. Do you want me to rename those tests to HP instead of openstack?